dangerzone/gui/main_window.py

```diff
@@ -402,6 +406,38 @@ def __init__(self) -> None:
         super(WaitingWidget, self).__init__()
 
 
+class TracebackWidget(QTextEdit):
+    """Reusable component to present tracebacks to the user.
+
+    By default, the widget is initialized but does not appear.
+    You need to call `.set_content("traceback")` on it so the
+    traceback is displayed.
+    """
+
+    def __init__(self) -> None:
+        super(TracebackWidget, self).__init__()
+        # Error
+        self.setReadOnly(True)
+        self.setStyleSheet(
+            """
+        QTextEdit {
+            font-family: Consolas, Monospace;
+            font-size: 12px;
+            background-color: #fff;
+            color: #000;
+            padding: 10px;
+        }
+        """
```

Comment:
Suggestion, let's move this to `dangerzone.css`, since it's kind of long and CSS related. A simple way to do this is to set a property to that widget:

```python
self.error_text.setProperty("style", "container_error")
```

and then add the following in `dangerzone.css`:

```diff
diff --git a/share/dangerzone.css b/share/dangerzone.css
index 45fdc2b..23f3bab 100644
--- a/share/dangerzone.css
+++ b/share/dangerzone.css
@@ -48,3 +48,11 @@ QLabel.version {
     font-size: 20px;
     padding-bottom: 5px;  /* align with 'dangerzone' font */
 }
+
+QTextEdit[style="container_error"] {
+    font-family: Consolas, Monospace;
+    font-size: 12px;
+    background-color: #ffffff;
+    color: #000000;
+    padding: 10px;
+}
```

I've tested already and it works.

Reply:
Awesome, thanks for the comment. I've updated the code and used the "traceback" CSS class name instead.